File: src/3079.find-the-sum-of-encrypted-integers.3367/solution.ts
function sumOfEncryptedInt (nums: number[]): number {
    let sum = 0;
    for (let i = 0; i < nums.length; i++) {
        sum += calc(nums[i]);
    }
    return sum;
}
const calc = (num:number) => {
    let max = 0;
    let base = 0;
    while (num > 0) {
        const digi = num % 10;
        max = Math.max(max, digi);
        base = base * 10 + 1;
        num = (num - digi) / 10;
    }
    return base * max;
};

File: src/3079.find-the-sum-of-encrypted-integers.3367/solution.java
class Solution {
    public int sumOfEncryptedInt(int[] nums) {
        int result = 0;
        for(int num : nums){
            result += encrypt(num);
        }
        return result;
    }

    private int encrypt(int num){
        int result = 0;
        int max = 0;
        while (num != 0){
            int digi = num%10;
            max = Math.max(max,digi);
            result = result*10+1;
            num /= 10;
        }
        return result*max;
    }

}